### PR TITLE
opengist: init at 1.6.1

### PR DESCRIPTION
**Pull request by lf-**
## Description of changes

This is a clone of github gist, with 100% more open source and 100% more self hosted.

https://github.com/thomiceli/opengist

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lf-** [comment]
@ofborg build opengist

File: pkgs/by-name/op/opengist/package.nix [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+buildGoModule {
+  pname = "opengist";
+  inherit version src;
+  vendorHash = "sha256-IorqXJKzUTUL5zfKRipZaJtRlwVOmTwolJXFG/34Ais=";
+  tags = [
+    "fs_embed"
+  ];
+
+  # required for tests
+  nativeBuildInputs = [
```
Comment (RaitoBezarius) [inline pkgs/by-name/op/opengist/package.nix:44]: `nativeCheckInputs` then?

File: pkgs/by-name/op/opengist/package.nix [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    git
+  ];
+
+  # required for tests
+  env = {
+    OG_OPENGIST_HOME = "/tmp/opengist";
```
Comment (RaitoBezarius) [inline pkgs/by-name/op/opengist/package.nix:50]: Shouldn't the /tmp be something like `$(mktemp -d)` ?

File: pkgs/by-name/op/opengist/package.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    inherit version src;
+
+    nativeBuildInputs = [ moreutils jq ];
+
+    # npm complains of "invalid package". shrug. we can give it a version.
+    preBuild = ''
+      jq '.version = "${version}"' package.json | sponge package.json
+    '';
+
+    # copy pasta from the Makefile upstream
```
Comment (RaitoBezarius) [inline pkgs/by-name/op/opengist/package.nix:22]: Can you put a link to the line of Makefile upstream?

**Review by RaitoBezarius — APPROVED** [review]
Looks good to me.
FYI, there's a little trick to avoid this `let` binding dance.
You can refer to `opengist` in the set of inputs of that package, i.e. `{ lib, buildGoModule, opengist, ... }:`

Then, you can set `version/src` normally without any let-binding and construct the frontend in-place inside the expr, then you refer to `inherit (opengist) version src;` to inherit from opengist itself.

This enables nice override properties without `finalAttrs` support (if I'm not wrong…).

**Comment by lf-** [comment]
> Looks good to me. FYI, there's a little trick to avoid this `let` binding dance. You can refer to `opengist` in the set of inputs of that package, i.e. `{ lib, buildGoModule, opengist, ... }:`
> 
> Then, you can set `version/src` normally without any let-binding and construct the frontend in-place inside the expr, then you refer to `inherit (opengist) version src;` to inherit from opengist itself.
> 
> This enables nice override properties without `finalAttrs` support (if I'm not wrong…).

Oh, evil :P I think I am going to leave it alone since I like more being able to tolerate the package being renamed, e.g.

https://github.com/NixOS/nixpkgs/issues/273815  🥺🥺🥺🥺

**Pull request merged**